vllm/attention/backends/hpu_attn.py

```diff
@@ -539,9 +541,29 @@ def forward(
             block_list = attn_metadata.block_list if attn_metadata \
                 and attn_metadata.block_list is not None else None
 
-            if self.sliding_window \
-               and attn_metadata.window_attn_bias is not None:
-                attn_bias = attn_metadata.window_attn_bias
+            common_args = self.common_attention_args(block_list, key_cache,
+                                                     value_cache,
+                                                     attn_metadata.block_size)
+
+            if self.sliding_window:
+                if attn_metadata.window_attn_bias is not None:
+                    attn_bias = attn_metadata.window_attn_bias
+
+                if self.use_fsdpa_window:
+                    slice_size = int(
+                        os.getenv("PT_HPU_QKV_SLICE_SEQ_LEN_THLD", "0"))
```

Comment:
suggest to move the initialization to __init__ func instead of checking env var in the middle of forwarding func